### ISSUE TITLE
Timelines: old topic gets removed immediately if retention_time = -1 (infinite)

### ISSUE THREAD

**Issue by eupestov** [zalando/nakadi#1121]
When a new timeline is created for an event type Nakadi sets the 'clean at' date for the previous timeline based on the event type retention time. After the retention period is over Nakadi deletes the old topic.
But if the event type's retention time is set to -1 the old topic is removed almost immediately because of the way the cleantAt date is calculated:

final Date cleanupDate = new Date(System.currentTimeMillis() + retentionTime);

https://github.com/zalando/nakadi/blob/96a6967fe88ac597ff0c0da54a5d3aa353a40e9c/src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java#L341

In our setup we use -1 for indefinite retention, although it is not described in Nakadi docs. So either Nakadi should not allow -1 for retention time or clean up should not happen.

**Comment by adyach** [comment]
@eupestov thank you for filing an issue.

just to clarify `-1` is a valid kafka `retention.ms` https://kafka.apache.org/documentation/#topicconfigs

**Comment by eupestov** [comment]
> just to clarify -1 is a valid kafka retention.ms

Yeah, that why we start using it. But is it to be officially supported by Nakadi? Taking into account it's future support for other messaging backends and other features like timelines.

**Comment by adyach** [comment]
> Yeah, that why we start using it. But is it to be officially supported by Nakadi? Taking into account it's future support for other messaging backends and other features like timelines.

it is gonna be discussed internally (I will update on that soon), from my point of view it makes, because kafka supports it.

**Comment by max-neverov** [comment]
hi, I'd like to work on this

**Comment by max-neverov** [comment]
PR https://github.com/zalando/nakadi/pull/1135
